http3/frames.go

```diff
@@ -131,3 +131,32 @@ func (f *settingsFrame) Write(b *bytes.Buffer) {
 		utils.WriteVarInt(b, val)
 	}
 }
+
+type goawayFrame struct {
+	Length   uint64
+	StreamID protocol.StreamID
+}
+
+func parseGoawayFrame(r io.Reader, l uint64) (*goawayFrame, error) {
+	buf := make([]byte, l)
+	if _, err := io.ReadFull(r, buf); err != nil {
+		if err == io.ErrUnexpectedEOF {
+			return nil, io.EOF
+		}
+		return nil, err
+	}
+	b := bytes.NewReader(buf)
```

Comment:
Reading the frame in a buffer and then reading from the buffer seems unnecessary.

Reply:
This is exactly how it was done in `parseSettingsFrame`, but I can look into alternatives